### PR TITLE
Integrate ECR dockercfg generator into this repository

### PR DESCRIPTION
**Pull request by mlocher**
This PR adds the functionality from https://github.com/codeship-library/aws-ecr-dockercfg-generator into this repository to make it easier to update all AWS related images to the newest version of the AWS CLI.

With the new configuration this repository will build three images `codeship/aws-base`, which is not pushed to the Docker Hub and the two previously built images `codeship/aws-deployment` and `codeship/aws-ecr-dockercfg-generator`, which are pushed to the Docker Hub.

I also added the _Code Of Conduct_ and a _CONTRIBUTING_ document (but I'd be happy to extract those to a different PR).

There will very likely be some changes that I missed, but I wanted to get feedback and a build on Codeship to see what's currently missing :)

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# Contributor Covenant Code of Conduct
```
Comment (rheinwein) [inline CODE_OF_CONDUCT.md]: ❤️

**Comment by rheinwein** [comment]
I'm cool with everything being in the same repo, makes it easier to test 👍 

@bfosberry any objections?

If not, looks good to me. The image name `codeship/aws-ecr-dockercfg-generator` stays the same, but we still need to change the links to the repos here: https://documentation.codeship.com/docker/dockercfg-services/#integrations

**Comment by mlocher** [comment]
From [Slack](https://codeship.slack.com/archives/dev/p1474395357001655)

> Brendan: @marko thats awesome as long as we are keeping both images as small as possible, which it looks like we are, so :+1:

But let's wait on more feedback directly on this ticket :)

**Comment by mlocher** [comment]
I added a PR to to the documentation to update the URL.

That said, in a next step I'd like to rename the repository to `aws` as the current name might be misleading with the added functionality from this PR (but this is just a heads up, we don't need to make that decision now)

**Comment by bfosberry** [comment]
LGTM!

**Comment by bfosberry** [comment]
We should update the dockercfg repo to redirect people here also

**Comment by mlocher** [comment]
@bfosberry I prepared a PR at https://github.com/codeship-library/aws-ecr-dockercfg-generator/pull/15 to redirect people from the old repository to this one.

**Pull request merged**